app/models/task.py

```diff
@@ -3,3 +3,29 @@
 
 class Task(db.Model):
     task_id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.String)
+    description = db.Column(db.String) 
+    completed_at = db.Column(db.DateTime, nullable=True)
+    goal_id = db.Column(db.Integer, db.ForeignKey('goal.goal_id'), nullable=True)
+    goal = db.relationship("Goal", back_populates="tasks")
+
+    def to_dict(self):
+        task_to_dict = {}
+        task_to_dict["id"] = self.task_id
+        task_to_dict["title"] = self.title
+        task_to_dict["description"] = self.description
+        # task_to_dict["completed_at"] = self.completed_at
+        if self.completed_at:
+            task_to_dict["is_complete"] = True
+        else:
+            task_to_dict["is_complete"] = False
+        
+        if self.goal_id:
+            task_to_dict["goal_id"] = self.goal_id
+
+        return task_to_dict
+
+    @classmethod
+    def from_dict(cls, task_data):
+        new_task = Task(title=task_data["title"], description=task_data["description"])
```

Comment:
This works! But what if we changed the name of the model? Now this would continue creating `Task` which no longer exists, so let's take advantage of the `cls` parameter. This will represent the current model, even if we change its name.

```python
        new_task = cls(title=task_data["title"], description=task_data["description"])
```